packages/data-context/src/data/ProjectLifecycleManager.ts

```diff
@@ -308,10 +308,20 @@ export class ProjectLifecycleManager {
   /**
    * Sets the initial `activeBrowser` depending on these criteria, in order of preference:
    *  1. The value of `--browser` passed via CLI.
-   *  2. The last browser selected in `open` mode (by name and channel) for this project.
-   *  3. The first browser found.
+   *  2. The value of `defaultBrowser` in `cypress.config`.
+   *  3. The last browser selected in `open` mode (by name and channel) for this project.
+   *  4. The first browser found.
    */
   async setInitialActiveBrowser () {
+    const configBrowser = this.loadedFullConfig?.defaultBrowser
+
+    // if we have a defaultBrowser from the config and a CLI browser wasn't passed,
+    // update the cliBrowser to the defaultBrowser from the config
+    // since we want the defaultBrowser to behave as if it was passed via CLI
+    if (configBrowser && !this.ctx.modeOptions.isBrowserGivenByCli) {
+      this.ctx.updateCliBrowser(configBrowser)
+    }
```

Comment:
Let's combine the conditionals and move `updateCliBrowser` to `BrowserActions.ts`.

```suggestion
    // if we have a defaultBrowser from the config and a CLI browser wasn't passed and,
    // we are in run mode or the CLI browser isn't set (prevents changing the browser in open mode when the config changes),
    // update the cliBrowser to the defaultBrowser from the config
    // since we want the defaultBrowser to behave as if it was passed via CLI
    const shouldUpdateCliBrowser = (configBrowser && this.ctx.modeOptions.isBrowserGivenByCli) &&
      (this.ctx.isRunMode || !this.ctx.coreData.cliBrowser)

    if (shouldUpdateCliBrowser) {
      this.ctx.actions.browser.updateCliBrowser(configBrowser)
    }
```

Reply:
resolved with latest commit